backend/src/utils/auth/authUtils.ts

```diff
@@ -0,0 +1,48 @@
+import jwt from 'jsonwebtoken';
+
+import {
+  DEFAULT_ACCESS_TIMEOUT,
+  DEFAULT_NO_SECRET,
+  DEFAULT_REFRESH_TIMEOUT,
+} from 'config/constants';
+import { IAccessJwtPayload, IRefreshJwtPayload, ITokens } from 'interfaces/auth/authInterfaces';
+import { IUser } from 'interfaces/db/entities';
+
+const generateJWT = async (userData: IUser): Promise<ITokens> => {
+  try {
+    if (!userData._id) {
+      throw new Error('wrong user id');
+    }
+
+    const { _id: id, role } = userData;
+    const timeout = process.env.JWT_EXPIRATION_TIME_ACCESS || DEFAULT_ACCESS_TIMEOUT;
+    const refreshTimeout = process.env.JWT_EXPIRATION_TIME_REFRESH || DEFAULT_REFRESH_TIMEOUT;
+
+    const accessToken = await jwt.sign({ id, role }, `${process.env.JWT_SECRET}`, {
+      expiresIn: timeout,
+    });
+
+    const refreshToken = await jwt.sign({ id }, `${process.env.JWT_REFRESH_TOKEN}`, {
+      expiresIn: refreshTimeout,
+    });
+    return { accessToken, refreshToken };
+  } catch (error) {
+    throw new Error('not generate token');
+  }
+};
+
+const verifyAccessToken = async (accessToken: string): Promise<IAccessJwtPayload> => {
+  const secret = process.env.JWT_SECRET || DEFAULT_NO_SECRET;
+
+  const isValid = (await jwt.verify(accessToken, secret)) as IAccessJwtPayload;
```

Comment:
As far as I know, jwt.verify method will throw an error, if verification can not be perfomed (for example if it has expired). So there is no need use isValid constant in that way. Correct me if I'm wrong 👀

Reply:
true, but this method also return an object when it's valid with decoded information of our token

Reply:
So `isValid` in that case represents not boolean value, but decoded object? If it does, It would be great to rename this varuable somehow to show what exactly it is used for.

Reply:
yes, maybe it's required some different name to better understanding, also you can see the interface that is attached to this object "IAccessJwtPayload" it's always will be a tip

Reply:
Yeeep, that is why I asked about renaming. The piece of code `as IAccessJwtPayload;` tells us about the final type of expression, but the name of varuable tells us the opposite.